consensus/spos/bls/subroundEndRound.go

```diff
@@ -131,6 +139,18 @@ func (sr *subroundEndRound) doEndRoundJob() bool {
 	return true
 }
 
+func (sr *subroundEndRound) signBlockHeader() ([]byte, error) {
+	headerCopy := sr.BlockProcessor().HeaderCopy(sr.Header)
+	headerCopy.SetLeaderSignature(nil)
+
+	marshalizedHdr, err := sr.Marshalizer().Marshal(headerCopy)
```

Comment:
or maybe a function inside the headerHandler interface which returns the header without leader signature.

Reply:
done

Reply:
but not without leader signature. maybe we'll use this function for other fields also